src/cascadia/TerminalApp/AppKeyBindings.cpp

```diff
@@ -1,117 +1,294 @@
-            case ShortcutAction::CloseTab:
-                _CloseTabHandlers();
-                return true;
-
-            case ShortcutAction::ScrollUp:
-                _ScrollUpHandlers();
-                return true;
-            case ShortcutAction::ScrollDown:
-                _ScrollDownHandlers();
-                return true;
-
-            case ShortcutAction::NextTab:
-                _NextTabHandlers();
-                return true;
-            case ShortcutAction::PrevTab:
-                _PrevTabHandlers();
-                return true;
-        }
-        return false;
-    }
-
-    // -------------------------------- Events ---------------------------------
-    DEFINE_EVENT(AppKeyBindings, CopyText,          _CopyTextHandlers,          TerminalApp::CopyTextEventArgs);
-    DEFINE_EVENT(AppKeyBindings, PasteText,         _PasteTextHandlers,         TerminalApp::PasteTextEventArgs);
-    DEFINE_EVENT(AppKeyBindings, NewTab,            _NewTabHandlers,            TerminalApp::NewTabEventArgs);
-    DEFINE_EVENT(AppKeyBindings, NewTabWithProfile, _NewTabWithProfileHandlers, TerminalApp::NewTabWithProfileEventArgs);
-    DEFINE_EVENT(AppKeyBindings, NewWindow,         _NewWindowHandlers,         TerminalApp::NewWindowEventArgs);
-    DEFINE_EVENT(AppKeyBindings, CloseWindow,       _CloseWindowHandlers,       TerminalApp::CloseWindowEventArgs);
-    DEFINE_EVENT(AppKeyBindings, CloseTab,          _CloseTabHandlers,          TerminalApp::CloseTabEventArgs);
-    DEFINE_EVENT(AppKeyBindings, SwitchToTab,       _SwitchToTabHandlers,       TerminalApp::SwitchToTabEventArgs);
-    DEFINE_EVENT(AppKeyBindings, NextTab,           _NextTabHandlers,           TerminalApp::NextTabEventArgs);
-    DEFINE_EVENT(AppKeyBindings, PrevTab,           _PrevTabHandlers,           TerminalApp::PrevTabEventArgs);
-    DEFINE_EVENT(AppKeyBindings, IncreaseFontSize,  _IncreaseFontSizeHandlers,  TerminalApp::IncreaseFontSizeEventArgs);
-    DEFINE_EVENT(AppKeyBindings, DecreaseFontSize,  _DecreaseFontSizeHandlers,  TerminalApp::DecreaseFontSizeEventArgs);
-    DEFINE_EVENT(AppKeyBindings, ScrollUp,          _ScrollUpHandlers,          TerminalApp::ScrollUpEventArgs);
-    DEFINE_EVENT(AppKeyBindings, ScrollDown,        _ScrollDownHandlers,        TerminalApp::ScrollDownEventArgs);
-
-
-}
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "AppKeyBindings.h"
+
+using namespace winrt::Microsoft::Terminal;
+using namespace winrt::TerminalApp;
+using namespace winrt::Windows::Data::Json;
+
+static const std::wstring KEYS_KEY{ L"keys" };
+static const std::wstring COMMAND_KEY{ L"command" };
+
+static const std::wstring COPYTEXT_KEY{ L"copy" };
+static const std::wstring PASTETEXT_KEY{ L"paste" };
+static const std::wstring NEWTAB_KEY{ L"newTab" };
+static const std::wstring NEWTABWITHPROFILE0_KEY{ L"newTabProfile0" };
+static const std::wstring NEWTABWITHPROFILE1_KEY{ L"newTabProfile1" };

```

Comment:
do we need `selectTab0` / `selectTab(0)` as well?

Reply:
Just merged master into the branch, added it now.